test/split/test_filter_header_rg.c

```diff
@@ -28,6 +28,59 @@ DEALINGS IN THE SOFTWARE.  */
 #include "../test.h"
 #include <unistd.h>
 
+int line_cmp(const void *av, const void *bv) {
+    const char *a = *(const char **) av;
+    const char *b = *(const char **) bv;
+    size_t al = strcspn(a, "\n");
+    size_t bl = strcspn(b, "\n");
+    size_t min = al < bl ? al : bl;
+    int m = memcmp(a, b, min);
+    if (m != 0) return m;
+    if (al < bl) return -1;
+    return al == bl ? 0 : 1;
+}
+
+bool hdrcmp(const char *hdr1, const char *hdr2) {
+    size_t nl1, nl2, count1 = 0, count2 = 0, i;
+    const char *l;
+    const char **lines1, **lines2;
+    int res = 0;
+
+    // First line should be @HD
+    if (strncmp(hdr1, "@HD\t", 4) != 0) return false;
+    if (strncmp(hdr2, "@HD\t", 4) != 0) return false;
+    nl1 = strcspn(hdr1, "\n");
+    nl2 = strcspn(hdr2, "\n");
+    if (nl1 != nl2 || memcmp(hdr1, hdr2, nl1) != 0) return false;
+
+    // Count lines.
+    for (l = hdr1 + nl1; *l != '\0'; l += strcspn(l, "\n")) ++l, ++count1;
+    for (l = hdr2 + nl2; *l != '\0'; l += strcspn(l, "\n")) ++l, ++count2;
+    if (count1 != count2) return false;
+    
+    lines1 = malloc(count1 * sizeof(*lines1));
+    if (!lines1) return false;
+    lines2 = malloc(count2 * sizeof(*lines2));
+    if (!lines2) return false;
+
+    for (i = 0, l = hdr1 + nl1; *l != '\0'; l += strcspn(l, "\n"))
```

Comment:
Are you skipping the first line intentionally? 
Currently, `lines1[0]` starts with the second line (`@SQ..`) and `lines1[2]` is an empty string.

To include the first line, we could have something like:
```
    for (i = 0, l = hdr1; *l != '\0'; l += strcspn(l, "\n")+1)
        lines1[i++] = l;
    for (i = 0, l = hdr2; *l != '\0'; l += strcspn(l, "\n")+1)
        lines2[i++] = l;
```
Otherwise, all the tests pass with the new header API and I happy to merge it.